addons/web/static/src/legacy/js/core/dialog.js

```diff
@@ -103,6 +103,7 @@ var Dialog = Widget.extend({
         this.renderHeader = options.renderHeader;
         this.renderFooter = options.renderFooter;
         this.onForceClose = options.onForceClose;
+        this.onPostDestroy = options.onPostDestroy;
```

Comment:
I'm not sure @odoo/rd-framework-js woud like that we add yet another hook here... unless they don't really care anymore because it's legacy and the hook would be supported in owl anyway ?

Reply:
Would listening for the `closed` event not work here? (or passing onForceClosed)

Reply:
Until the modal element is in the dom with a `tabindex` attribute being set or is visible within the dom (`display: block;`), the `link.focus()` does not work.
There is no hooks after `this.$modal.css('display', 'none');` or `this.$modal.remove();`.

@sdegueldre Do you have another suggestion ?

Reply:
Since all the destroy code is synchronous you can schedule the focus in the next microtick:
```js
Promise.resolve().then(() => link.focus());
```
And that should work.

Reply:
@sdegueldre The reason I did not use this kind of strategy is because it is not reliable and make the code unclear.
I largely prefer a hook solution.

Reply:
I agree with @Goaman. Any other alternative you might have in mind ?

Reply:
Hello @Goaman @dmo-odoo 

I'm not sure about adding another hook/param to the dialog API, but I agree the Promise.resolve/setTimeout thing is a hack. However, I'm sure that the timing is not good, as we're currently rewriting everything in owl. What about keeping the hack for now, and reconsidering introducing something new in the (wowl) dialog API when this code will be converted?

Reply:
This sounds acceptable to me.

Reply:
@aab-odoo 
What is the problem of adding a hook/param ?
Why not adding the hook and removing it when everything is written in owl.
Especially that the hook is a reminder that there is some code that need to be called at that time.

Reply:
I've removed the hook and used the microtask strategy with a comment.